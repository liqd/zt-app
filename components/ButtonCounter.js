import React from 'react';
import { Text, TouchableOpacity } from 'react-native';
import { styles } from './ButtonCounter.styles';

export const ButtonCounter = (props) => {
  return (
    <TouchableOpacity
      style={styles.container}
      onPress={props.onPress}
      disabled={props.disabled}
    >
      <Text style={
        props.disabled
          ? styles.textDisabled
          : !props.highlight || props.highlight === 0
            ? styles.text
            : props.highlight === 1
              ? styles.textUpHighlight
              : styles.textDownHighlight
      }>
        {props.counter}
      </Text>
      <Text style={
        props.disabled
          ? styles.iconDisabled
          : !props.highlight || props.highlight === 0
            ? styles.icon
            : props.highlight === 1
              ? styles.iconUpHighlight
              : styles.iconDownHighlight
      }>
        {props.icon}
      </Text>
    </TouchableOpacity>
  );
};
