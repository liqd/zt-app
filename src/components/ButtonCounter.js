import React from 'react'
import { Pressable, View } from 'react-native'

import { styles } from './ButtonCounter.styles'
import { TextSourceSans } from './TextSourceSans'

export const ButtonCounter = (props) => {
  const listStyle = [styles.baseText, styles.disableText, styles.listText]
  let btnStyle
  let textStyle
  if (props.disabled) {
    btnStyle = styles.btn
    textStyle = [styles.baseText, styles.disableText]
  } else {
    if (props.counter === 0) {
      btnStyle = styles.btn
      textStyle = styles.baseText
    } else {
      if (props.rating === 'pos') {
        btnStyle = styles.btn
        textStyle = [styles.baseText, styles.ratedUp]
        if (props.highlight) {
          btnStyle = [styles.btn, styles.highlightUp]
          textStyle = [styles.baseText, styles.ratedUp]
        }
      } else if (props.rating === 'neg') {
        btnStyle = styles.btn
        textStyle = [styles.baseText, styles.ratedDown]
        if (props.highlight) {
          btnStyle = [styles.btn, styles.highlightDown]
          textStyle = [styles.baseText, styles.ratedDown]
        }
      }
    }
  }

  return (props.list ?
    <View style={styles.container}>
      <TextSourceSans style={listStyle}>
        {props.counter}
      </TextSourceSans>
      <TextSourceSans style={listStyle}>
        {props.icon}
      </TextSourceSans>
    </View>
    :
    <View style={styles.container}>
      <Pressable
        style={btnStyle}
        onPress={props.onPress}
        accessibilityLabel={props.labelText}
        accessibilityHint={props.hintText}
        disabled={props.disabled}
      >
        <TextSourceSans style={textStyle}>
          {props.counter}
        </TextSourceSans>
        <TextSourceSans style={textStyle}>
          {props.icon}
        </TextSourceSans>
      </Pressable>
    </View>
  )
}
