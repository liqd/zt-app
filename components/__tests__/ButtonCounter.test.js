import React from 'react'
import Icon from 'react-native-vector-icons/SimpleLineIcons'
import { render } from '@testing-library/react-native'

import { ButtonCounter } from '../ButtonCounter'

test('Test ButtonCounter', () => {
  const { getByText, toJSON } = render(
    <ButtonCounter
      disabled={false}
      highlight={0}
      counter={0}
      icon={<Icon name="check" />}
    />
  )
  expect(toJSON()).toMatchSnapshot()
  expect(getByText('0')).toBeTruthy()
})

test('Test ButtonCounter 10', () => {
  const { getByText, toJSON } = render(
    <ButtonCounter
      disabled={false}
      highlight={0}
      counter={10}
      icon={<Icon name="check" />}
    />
  )
  expect(toJSON()).toMatchSnapshot()
  expect(getByText('10')).toBeTruthy()
})

test('Test ButtonCounter random value', () => {
  const rand = Math.floor(Math.random() * 100)
  const { getByText } = render(
    <ButtonCounter
      disabled={false}
      highlight={0}
      counter={rand}
      icon={<Icon name="check" />}
    />
  )
  expect(getByText(`${rand}`)).toBeTruthy()
})

test('Test ButtonCounter disabled', () => {
  const { getByText, toJSON } = render(
    <ButtonCounter
      disabled={true}
      highlight={0}
      counter={0}
      icon={<Icon name="check" />}
    />
  )
  expect(toJSON()).toMatchSnapshot()
  expect(getByText('0')).toBeTruthy()
})

test('Test ButtonCounter highlight', () => {
  const { getByText, toJSON } = render(
    <ButtonCounter
      disabled={false}
      highlight={1}
      counter={0}
      icon={<Icon name="check" />}
    />
  )
  expect(toJSON()).toMatchSnapshot()
  expect(getByText('0')).toBeTruthy()
})

test('Test ButtonCounter highlight 10', () => {
  const { getByText, toJSON } = render(
    <ButtonCounter
      disabled={false}
      highlight={10}
      counter={0}
      icon={<Icon name="check" />}
    />
  )
  expect(toJSON()).toMatchSnapshot()
  expect(getByText('0')).toBeTruthy()
})
