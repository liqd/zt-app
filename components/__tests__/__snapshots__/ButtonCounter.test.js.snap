// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ButtonCounter 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "opacity": 1,
      "paddingVertical": 8,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "SourceSansPro_400Regular",
        },
        Object {
          "color": "#1a1a1a",
          "letterSpacing": 5,
        },
      ]
    }
  >
    0
  </Text>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "SourceSansPro_400Regular",
        },
        Object {
          "color": "#1a1a1a",
          "fontSize": 20.4,
          "letterSpacing": 10,
          "lineHeight": 24.200000000000003,
        },
      ]
    }
  >
    <Text />
  </Text>
</View>
`;

exports[`Test ButtonCounter 10 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "opacity": 1,
      "paddingVertical": 8,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "SourceSansPro_400Regular",
        },
        Object {
          "color": "#1a1a1a",
          "letterSpacing": 5,
        },
      ]
    }
  >
    10
  </Text>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "SourceSansPro_400Regular",
        },
        Object {
          "color": "#1a1a1a",
          "fontSize": 20.4,
          "letterSpacing": 10,
          "lineHeight": 24.200000000000003,
        },
      ]
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          undefined,
          Object {
            "fontFamily": "simple-line-icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </Text>
</View>
`;

exports[`Test ButtonCounter disabled 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "opacity": 1,
      "paddingVertical": 8,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "SourceSansPro_400Regular",
        },
        Object {
          "color": "#c4c4c4",
          "letterSpacing": 5,
        },
      ]
    }
  >
    0
  </Text>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "SourceSansPro_400Regular",
        },
        Object {
          "color": "#c4c4c4",
          "fontSize": 20.4,
          "letterSpacing": 10,
          "lineHeight": 24.200000000000003,
        },
      ]
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          undefined,
          Object {
            "fontFamily": "simple-line-icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </Text>
</View>
`;

exports[`Test ButtonCounter highlight 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "opacity": 1,
      "paddingVertical": 8,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "SourceSansPro_400Regular",
        },
        Object {
          "color": "#2d40cc",
          "fontWeight": "bold",
          "letterSpacing": 5,
        },
      ]
    }
  >
    0
  </Text>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "SourceSansPro_400Regular",
        },
        Object {
          "color": "#2d40cc",
          "fontSize": 20.4,
          "fontWeight": "bold",
          "letterSpacing": 10,
          "lineHeight": 24.200000000000003,
        },
      ]
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          undefined,
          Object {
            "fontFamily": "simple-line-icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </Text>
</View>
`;

exports[`Test ButtonCounter highlight 10 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "opacity": 1,
      "paddingVertical": 8,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "SourceSansPro_400Regular",
        },
        Object {
          "color": "#a72b1e",
          "fontWeight": "bold",
          "letterSpacing": 5,
        },
      ]
    }
  >
    0
  </Text>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "SourceSansPro_400Regular",
        },
        Object {
          "color": "#a72b1e",
          "fontSize": 20.4,
          "fontWeight": "bold",
          "letterSpacing": 10,
          "lineHeight": 24.200000000000003,
        },
      ]
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          undefined,
          Object {
            "fontFamily": "simple-line-icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </Text>
</View>
`;
